update_image.pl

```diff
@@ -375,24 +387,27 @@ sub processConvergedSections {
     {
         if(-e $ocmbfw_original_filename)
         {
-            #Verify image validity
-            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename");
-            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename";
+            #Add header to delivered image
+            my $date = `date`;
+            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $ocmbfw_original_filename --packagedBin $ocmbfw_original_filename.header --timestamp \"$date\" --vendorVersion \"$ocmbfw_version\" --vendorUrl \"$ocmbfw_url\"");
+
+            #Verify header sha512 hash value matches value calculated against image
+            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename.header");
+            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename.header";
         }
         else
         {
             print "WARNING: OCMBFW binary not found, generating blank binary (w/ valid header) instead\n";
-            #Create blanke 4k image
-            my $ocmbfw_generated_filename = "ocmbfw_generated.bin";
-            run_command("dd if=/dev/zero of=$scratch_dir/$ocmbfw_generated_filename bs=1024 count=4");
+            #Create blank 4k image
+            run_command("dd if=/dev/zero of=$ocmbfw_original_filename bs=1024 count=4");
         
             #Add header to blank image
             my $date = `date`;
-            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $scratch_dir/$ocmbfw_generated_filename --packagedBin $ocmbfw_original_filename --timestamp \"$date\" --vendorVersion \"0.1\" --vendorUrl \"http://www.ibm.com\"");
+            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $ocmbfw_original_filename --packagedBin $ocmbfw_original_filename.header --timestamp \"$date\" --vendorVersion \"$ocmbfw_version\" --vendorUrl \"$ocmbfw_url\"");
 
             # verify header sha512 hash value matches value calculated against image
-            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename");
-            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename";
+            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename.header");
+            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename.header";
         }
         #Final image will be under a new name after ECC protection and any other processing required
         $sections{OCMBFW}{out}       = "$ocmbfw_binary_filename";
```

Comment:
can you please remind me again why this line is different then the previous $sections{<pnor_partition>}{out} lines like this: 
$sections{WOFDATA}{out} = "$scratch_dir/wofdata.bin.ecc"

I vaguely recall that there was something special for this and why we didn't need to at least use the $scratch_dir part of it.

Reply:
Because the scratch dir is passed in as part of the overall ocmbfw_original_filename:

https://github.com/open-power/op-build/blob/master/openpower/package/openpower-pnor/openpower-pnor.mk#L132

Reply:
great - thanks fore reminder